### ISSUE TITLE
Oscillation of territory estimation

### ISSUE THREAD

**Issue by kaorahi** [lightvector/KataGo#13]
Oscillation of territory estimation is observed in a variation of leela zero
(https://github.com/leela-zero/leela-zero/issues/2331#issuecomment-483219289).
Does KataGo have a similar tendency?

**Comment by lightvector** [comment]
@ihavnoid - Is that the raw neural net output or is that averaged across the search? I usually only look at the value averaged across the search (in exactly the same way as winrate), in which case there is no noticeable oscillation - the final score estimate is pretty good. In an individual position, the raw neural net may oscillate a little (although I'd be a little surprised if it was always anywhere near that bad), but the final MCTS output is stable and very good.

Note also that I've generally found that summing up a final ownership/endstate estimate does *not* always produce as good of an estimate of the final score as if you train a single output directly to predict the final score. It will still be very good, but to get a crude sort of intuition as to why - imagine you took a board position where a human pro was correctly confident that one side was currently ahead by about 5 points, but not much more, and neither much less (not just in current territory, but also taking into account future moyo and potential and followups). Now you imagine instead of doing that, you asked the pro to separately on every board coordinate (laboriously) give you a literal numeric percentage chance for black vs white to own that specific point, and and with each prediction focusing only on getting that prediction right without specific care to get the *sum* of them right. And then you add up all their numeric answers. Will that also come out to also precisely 5 points? For humans, actually the process of making all those separate predictions and making up numeric percentages will add noise and possible bias and it will not really come out as good when you add them up. 

Neural nets are not nearly as bad at being wholistically consistent, but the same intuition carries over. Another way of thinking about it is when you train to predict the final result point-by-point, if you have no penalty in the loss function specifically aiming to get their sum correct, only each individual point, the sum may end up generally having somewhat higher noise in the final result than if the neural net also cared specifically about the sum.

**Comment by kaorahi** [comment]
Thanks for explanation. The oscillation plot in the above link is the raw neural net output, I think.

This version of LZ tends to be optimistic in my observation. The estimation seems biased to black in black's turn, and to white in white's turn.

**Comment by ihavnoid** [comment]
Yeah, I didn't even implement how to print out the territory (it just exists internally) so it really depends on whoever implemented their way of printing out the territory information.  I have no clear reason why it is trained to be consistently optimistic, though.
